Cabal/Distribution/PackageDescription/Parse.hs

```diff
@@ -1213,14 +1223,14 @@ parseHookedBuildInfo inp = do
           lowercase inFieldName /= "library" &&
           lowercase inFieldName /= "benchmark" &&
           lowercase inFieldName /= "test-suite"
-            = liftM (\bis -> [(CLibName "", bis)]) (parseBI bi)
+            = liftM (\bis -> [(CLibName, bis)]) (parseBI bi)
     parseLib _ = return []
 
     parseStanza :: HookedBuildInfo -> [Field] -> ParseResult HookedBuildInfo
     parseStanza bis (F line inFieldName mName:bi)
         | Just k <- case lowercase inFieldName of
                         "executable" -> Just CExeName
-                        "library"    -> Just CLibName
+                        "library"    -> Just CSubLibName
```

Comment:
`HookedBuildInfo` really needs to be better documented...
Also looks like `isStanzaHeader` should be made aware of test suites and benchmarks, right now it assumes that "executable" is the only non-library stanza type.